app/src/org/commcare/dalvik/application/AndroidShortcuts.java

```diff
@@ -32,23 +34,32 @@ protected void onCreate(Bundle bundle) {
         final Intent intent = getIntent();
         final String action = intent.getAction();
 
-        //The Android needs to know what shortcuts are available, generate the list
+        // The Android needs to know what shortcuts are available, generate the list
         if (Intent.ACTION_CREATE_SHORTCUT.equals(action)) {
             buildMenuList();
         }
     }
-    
-    private void buildMenuList() {
-        ArrayList<String> names = new ArrayList<String>();
-        ArrayList<String> commands = new ArrayList<String>();
-        
-        AlertDialog.Builder builder = new AlertDialog.Builder(this);
-        
-        builder.setTitle("Select CommCare Shortcut");
 
-        for(Suite s : CommCareApplication._().getCommCarePlatform().getInstalledSuites()) {
-            for(org.commcare.suite.model.Menu m : s.getMenus()) {
-                if("root".equals(m.getRoot())) {
+    private void buildMenuList() {
+        final PaneledChoiceDialog dialog = new PaneledChoiceDialog(this, "Select CommCare Shortcut");
+        dialog.setChoiceItems(getChoiceItemList(dialog));
+        dialog.setOnCancelListener(new OnCancelListener() {
+            public void onCancel(DialogInterface dialog) {
+                AndroidShortcuts sc = AndroidShortcuts.this;
+                sc.setResult(RESULT_CANCELED);
+                sc.finish();
+                return;

```

Comment:
@phillipm I think sometimes a return is actually needed after calling finish(). I'm not at all versed in when that is the case, but this is code that I just moved from one place to another so I think I'm going to leave it the way it was.